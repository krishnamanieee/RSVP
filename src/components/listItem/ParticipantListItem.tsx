import React from 'react';
import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import Theme from '../../theme';

interface IParticipantListItemProps {
  index: number;
  name: string;
  locality: string;
  onPress: () => void;
}

class ParticipantListItem extends React.PureComponent<
  IParticipantListItemProps,
  any
> {
  render(): React.ReactElement {
    const {index, name, locality, onPress} = this.props;
    return (
      <TouchableOpacity
        onPress={onPress}
        style={[
          styles.container,
          {
            backgroundColor:
              index % 2 === 0 ? Theme.colors.white : Theme.colors.lightGray,
          },
        ]}>
        <View style={{flexDirection: 'row', alignItems: 'center'}}>
          <View style={styles.profileContainer}>
            <Text style={styles.profileText}>
              {name.substring(0, 2).toUpperCase()}
            </Text>
          </View>
          <View style={{padding: 10}}>
            <Text style={{paddingVertical: 5}}>{name}</Text>
            <Text style={styles.localityText}>
              <Text style={{fontWeight: 'bold'}}>Locality: </Text>
              {locality}
            </Text>
          </View>
        </View>
      </TouchableOpacity>
    );
  }
}

export default ParticipantListItem;

const styles = StyleSheet.create({
  container: {
    padding: 10,
  },
  profileContainer: {
    height: 50,
    width: 50,
    borderRadius: 25,
    backgroundColor: Theme.colors.primaryColor,
    justifyContent: 'center',
    alignItems: 'center',
  },
  profileText: {
    color: Theme.colors.white,
    fontSize: Theme.font.fontSize.large,
    fontWeight: 'bold',
  },
  localityText: {
    fontSize: Theme.font.fontSize.small,
  },
});
