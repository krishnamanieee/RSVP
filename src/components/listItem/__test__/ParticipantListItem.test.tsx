import React from 'react';
import renderer from 'react-test-renderer';
import ParticipantListItem, {
  IParticipantListItemProps,
} from '../ParticipantListItem';

let props: IParticipantListItemProps = {
  index: 0,
  locality: '',
  name: '',
  onPress: jest.fn,
};
describe('ParticipantListItem', () => {
  it('should match match snapshot', () => {
    const tree = renderer.create(<ParticipantListItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
