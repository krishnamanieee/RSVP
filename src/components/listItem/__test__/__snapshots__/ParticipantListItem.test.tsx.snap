// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantListItem should match match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#da2c56",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "width": 50,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "paddingVertical": 5,
          }
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Locality: 
        </Text>
        
      </Text>
    </View>
  </View>
</View>
`;
